### ISSUE TITLE
force maskrom mode?

### ISSUE THREAD

**Issue by colemickens** [colemickens/h96-max-v58-investigation#4]
Unfortunately, I think I put a bad u-boot on here, so now I can't boot to `rkusb` mode in u-boot.

But, there's enough there that maskrom mode doesn't boot, and it loads u-boot.

So... it would appear I need to figure out where the pins to FORCE maskrom mode on this board are.

And I have no idea. :)

**Comment by colemickens** [comment]
Alternatively, I could maybe hack something with the existing u-boot on there (it does boot).

But alas, I can't send input over serial... not sure if it's a u-boot issue, or if there's some bridging/pull-down I need to do??

**Comment by samueldr** [comment]
Your bad U-Boot is on eMMC, right?

Find out the pinout and figure out if there's a way to "mask" it from the system, something like tying its CLK to GND, or tying a chip select line to the opposite of "selected".

**Comment by colemickens** [comment]
> Your bad U-Boot is on eMMC, right?

Yep - yeah, thanks, I roughly get that's the goal, sadly my electronics+electricity understanding is such that I don't know how to determine what those are. The eMMC is "in" the PCB in a way that I can't even see the traces, even if I did have a diagram.

There are some suspicious looking pairs of pads on the board right next to the emmc (two pairs on top, one pair on bottom) that I'm attempting to short, but I'm also running into other issues - its unclear if I need to apply board power or just USB, my serial adapter is resetting at really inconvenient, my internet is bad enough that its hard to tell if SSH went out or serial dropped, the board seems to leach power and stay in weird power states at times, etc.

I'm going to throw a hub in the mix and/or try another host to connect from, to see if that gets the serial adapter to play nice again. I don't recall having this issue months ago when the "flasher" was my regular laptop -- right now "rocky5b" is the  serial monitor-er and would-be flasher, and who knows what USB bugs the rock5b 5.10 vendor kernel may contain.

Oh, and I'm doing this via a (dutiful, helpful!) remote operator, and it's all a bit more challenging that way (though I did get `rocky5b` and `visionfive2` back up and running this way, complete with some trips to bootrom modes). 

(off topic brag:) (I actually even fixed vf2 remotely by building a "thin" systemd-initrd-enabled netboot image that downloads squashfs of stage-2 /nix/store during stage-1. So it boots to Linux fast, but still can drop you in a nice environment with lots of beefy packages... had to brag a bit, it's quite handy to have if one can at least get ethernet up under u-boot! Even have a little script that `wtype`s the `tftpboot` commands into the u-boot serial connection, since pasting directly wasn't working!)

**Comment by samueldr** [comment]
Are there good resolution pictures of all board sides?

**Comment by colemickens** [comment]
eh, this is what I have so far:
![IMG_1781](https://github.com/colemickens/h96-max-v58-investigation/assets/327028/b673bd34-0ba8-4b08-8dfb-fc5daceb7c7e)
![IMG_1780](https://github.com/colemickens/h96-max-v58-investigation/assets/327028/05638801-867c-4c01-b648-d3fd2d3cae8a)

**Comment by samueldr** [comment]
(Any *you* would not apply currently since you need remote hands to *hand*le.)

A lot of assumptions, rambling, and supposition incoming.

I see, BGA, `KLMC64JENB`. Fun thing: their markings show the grid (A through P, then 1 through 14). Not that it really helps since those are not directly exposed. The interface AFAICT is FBGA153, standard for eMMC.

Though it's likely passives on the other side are related to it, there's no trivial way to check that I know.

![image](https://github.com/colemickens/h96-max-v58-investigation/assets/132835/e7691f55-a9c0-46ba-bc03-cb2fbe5e1445)

Given the amount of `NC` and `RFU`, I'd think it's plausible you'll find passed-through on the other side the RST_n, or CLK signals. Given *incorrect* measurements with my mouse and the picture, there is a cluster of passive right about where it would be expected. You might be able to guess which would be linked to CLK since you maybe can suss out Vssq Vccq, and other repeated pins via continuity. Though that's really grasping at straws.

Still grasping at straws, there are tiny black passives on squiggly lines, 10 of them, around the area, spread on the two sides. Squiggly lines are a sign of, uh, length matching(?)... There are 8 DAT lines, one CMD and one CLK. Even more grasping at straws, but those look like plausible options.

I tried deforming and aligning your images, and overlaying them, rough draft of the emmc section here:

![image](https://github.com/colemickens/h96-max-v58-investigation/assets/132835/43cb47dd-f0b1-44f3-9a04-184c1ce89bd7)

I've highlighted the 10 sus lines' passives.

These maps to:

Back:

![image](https://github.com/colemickens/h96-max-v58-investigation/assets/132835/f730204b-1d83-4164-ade0-a3400790399d)

Front:

![image](https://github.com/colemickens/h96-max-v58-investigation/assets/132835/954e33c1-29e7-4a58-81c9-014a718d6cf1)

**Comment by colemickens** [comment]
somehow I found it... zoom, zoom, zoom
![PXL_20230828_053311940](https://github.com/colemickens/h96-max-v58-investigation/assets/327028/467a9e4b-b616-4200-9736-3ce8b4f40954)

now to iterate on this tow-boot/u-boot build]

**Comment by SzymonZy** [comment]
![obraz](https://github.com/colemickens/h96-max-v58-investigation/assets/3253414/2c560b71-4bec-4e0c-b404-e2b0c6cbe20d)
Unplug usb, nvme. Connect these two pins and power connect usb A-A cable to usb 2.0 port. It is tricky and I did it with small twizzers. You will get into maskrom and you vid/pid will be generic rockchip maskrom device. I think it was well known method since January.  You can flash later from linux any image into eMMC. At least I did that half of a year ago. There is a guy THC13 no Radxa discord who played with me a lot and shown that.

**Comment by colemickens** [comment]
:pray: thanks! if one zooms in really far on the circle in my image, shorting it to ground also works 100% of the time. For me it was even easier because I can clip onto any ground point and then only need to carefully position one connection. For me, holding just one spot and connecting power was hard enough with only two hands :laughing:.

Now I just need to figure out the u-boot build... so far I haven't managed to get it to actually initialize usb or networking enough to get a boot going. Though it seems that the rock5b defconfig gets close.

> There is a guy THC13 no Radxa discord who played with me a lot and shown that.

Maybe I should circle back to the Discord and see if anyone else is hacking on u-boot for the board.
